src/core/tritonserver.cc

```diff
@@ -2414,9 +2438,9 @@ TRITONSERVER_ServerInferAsync(
   // object associated with the inference request above.
 #ifdef TRITON_ENABLE_TRACING
   if (!status.IsOk()) {
-    std::unique_ptr<ni::InferenceTrace>* trace = ureq->MutableTrace();
+    std::shared_ptr<ni::InferenceTraceProxy>* trace = ureq->MutableTrace();
     if (*trace != nullptr) {
-      ni::InferenceTrace::Release(std::move(*trace));
+      *trace = nullptr;
```

Comment:
Need to add a InferenceRequest::ReleaseTrace() function and have that release both the InferenceRequest and the InferenceResponseFactory trace object.

Reply:
change to call ureq->ReleaseTrace()